Noah assortedbugfixes

noahvaks opened the pull request:
daily sales report sort by membership type
daily sales report has number of executive members and regular members
daily sales report has total daily revenue
daily sales report include member name
daily sales report sets decimals 

add sales tax to receipts and daily sales reports

pos combo box doesn't update on item changes
pos combo box doesn't update on member changes

add headers for pages

suggested upgrades and downgrades by administrator access only

noahvaks committed:
bugfixes after saturday run through

daily sales report filters by membership type
daily sales report lists number of each member type
daily sales report lists daily revenue
daily sales report includes member name
daily sales report proper number of decimals

add sales tax to POS receipt and daily sales report

POS combo boxes update after adding/deleting items and members

added headers for every page

suggested upgrades and downgrades now only accessible to administrator

simonisfiction commented:
tested by simon. Ii found one issue. if sally shopper happens to buy 4 things on one day, then she will add 4 to the count of members that shopped that day
she should only add 1 to the count

noahvaks committed:
count distinct members 

error on counting executive members on 3/13

Dogguy0210 committed:
fixed bug that occured no excutive members made a purchase on a given
day, and added a work around for the bug caused by deleting and item or
member (cascade is not working).

Dogguy0210 committed:
Got rid an unessary qDebug() that may or may not of been causing a bug

bigverl commented:
Checked "Daily Sales Report" counting of executive and regular members. Looks good.

Dogguy0210 committed:
Merge branch 'development' into noah-assortedbugfixes
